stripe-ui-core/src/test/java/com/stripe/android/uicore/elements/TextFieldTest.kt

```diff
@@ -0,0 +1,203 @@
+package com.stripe.android.uicore.elements
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.R
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.semantics.SemanticsActions
+import androidx.compose.ui.semantics.SemanticsProperties.TextSelectionRange
+import androidx.compose.ui.test.SemanticsMatcher
+import androidx.compose.ui.test.SemanticsNodeInteraction
+import androidx.compose.ui.test.assert
+import androidx.compose.ui.test.hasInsertTextAtCursorAction
+import androidx.compose.ui.test.hasRequestFocusAction
+import androidx.compose.ui.test.hasSetTextAction
+import androidx.compose.ui.test.isEnabled
+import androidx.compose.ui.test.isFocused
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performSemanticsAction
+import androidx.compose.ui.test.performTextInput
+import androidx.compose.ui.text.TextRange
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardCapitalization
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class TextFieldTest {
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    @Test
+    fun `On initial value passed, selection should be at the start of the field`() {
+        composeTestRule.setContent {
+            TestTextField(initialValue = "A1B")
+        }
+
+        composeTestRule
+            .onNodeWithTag(TEST_TAG)
+            .assert(hasTextSelection(TextRange(3)))
+    }
+
+    @Test
+    fun `On input, selection should move with input`() {
+        composeTestRule.setContent {
+            TestTextField(initialValue = "A1B")
+        }
+
+        val textField = composeTestRule.onNodeWithTag(TEST_TAG)
+
+        textField.performTextInput("2")
+        textField.performTextInput("C")
+
+        textField.assert(hasTextSelection(TextRange(5)))
+    }
+
+    @Test
+    fun `On input in middle of text, selection should move with input`() {
+        composeTestRule.setContent {
+            TestTextField(initialValue = "A1B")
+        }
+
+        val textField = composeTestRule.onNodeWithTag(TEST_TAG)
+
+        textField.performTextInputSelection(TextRange(1))
+
+        textField.performTextInput("2")
+        textField.performTextInput("C")
+
+        textField.assert(hasTextSelection(TextRange(3)))
```

Comment:
should we also verify the text at this point? ie would the text field now contain "A2C"? I think that would just be useful so that we can more easily understand expected behavior of the text field from looking at this test